### PR TITLE
Attach Preallocated buffers to MIOpen for the Find process

### PR DESCRIPTION
**Pull request by umangyadav**
Changes as per  this PR from MIOpen : https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1839

### CODE REVIEW

**Comment by umangyadav** [comment]
Need to update this PR based on https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1839

**Comment by umangyadav** [comment]
@pfultz2 @DrizztDoUrden can you guys take a look at this ? MIGraphX CI is using ROCm 5.5, so this still wouldn't get tested inside CI,  upon testing locally with ROCm 5.6 it passes.

**Comment by pfultz2** [comment]
You should do a `check_library_exists` for `miopenSetFindOptionPreallocatedTensor` function instead of checking miopen versions.

File: src/targets/gpu/CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,8 @@
 list(APPEND CMAKE_PREFIX_PATH /opt/rocm)
 find_package(miopen)
 
+configure_file(${CMAKE_SOURCE_DIR}/src/version.h.in ${CMAKE_BINARY_DIR}/src/include/migraphx/version.h)
```
Comment (pfultz2) [inline src/targets/gpu/CMakeLists.txt:28]: Remove this, the file is already configured in src/CMakeLists.txt.

File: src/version.h.in [hunk 2]
```diff
@@ -26,4 +26,7 @@
 #define MIGRAPHX_VERSION_MINOR @PROJECT_VERSION_MINOR@
 #define MIGRAPHX_VERSION_PATCH @PROJECT_VERSION_PATCH@
 #define MIGRAPHX_VERSION_TWEAK @PROJECT_VERSION_TWEAK@
+#define MIGRAPHX_MIOPEN_VERSION_MAJOR @miopen_VERSION_MAJOR@
+#define MIGRAPHX_MIOPEN_VERSION_MINOR @miopen_VERSION_MINOR@
+#define MIGRAPHX_MIOPEN_VERSION_PATCH @miopen_VERSION_PATCH@
```
Comment (pfultz2) [inline src/version.h.in:31]: Remove these defines, They dont make sense when not building gpu backend.

Comment (umangyadav) [reply src/version.h.in]: Done.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**